### ISSUE TITLE
Lets formulate a required CI/support for the projects

### ISSUE THREAD

**Issue by Anton-Latukha** [haskell-nix/hnix-store#112]
Since the projects depend on each other - what Store supports directly influences the downstream (HNix, DHall-Nix and our projects).

I put these dialog questions:
1. We must decide what to do with macOS support.
<details><summary>My previous yapping about it </summary>
<p>

> 
> And macOS platform is changing, we know.
> 
> Now macOS platform is already in the transition. Nixpkgs and we - would need to support both for quite some time. And after the transition, it is obvious it would be more tighter and tighter in a proprietary way, and more and more different from Linux.
> 
> My view is: I personally, do not know specifics of, or own any macOS devices. sorki most probably the same way.
> But I agree that macOS is a big platform with a lot of devs, Haskell devs & some Nix people sitting on it. John AFAIK used macOS.
> I think the bare minimum - we should test & support classical Haskell development for the projects on macOS. So HLS & Hackage & Cabal development works on macOS. That is a big requirement and a minimum requirement and at the same time - a very good grade of & classical development workflow provided. we can supply to the platform.
> 
> The Nix dev env on macOS depends on Nixpkgs situations & support of it, & IDK what happens and would be happening there. I remember Dolstra was taking about dropping Hydra cache/builds of it. We probably can/should support macOS in Nix, but it would've been great if there were people who were interested in refactoring the projects to support it & maintaining the software stack for the platform. I personally (at all) not very interested in Nix dev env on macOS.
> 
> But whatever choice of support we do - we must do it together. I propose to introduce a Hackage Cabal macOS test. But about Nix dev env I would tag along for any team decision and would stay unaffectionate about the Nix dev env for macOS.
> I'd at the very minimum deferred the Nix dev env for macOS from the PR check to the master scheduled check. Because PR checks soft of pressure the devs and exposes infra side effects unrelated to work they provide. We all want CI for our working patches to be reaffirm with the color green. And probably'd removed the Nix dev env for macOS CI check fully, until someone would be interested in it. HLS + Hackage Cabal macOS workflow if fully enough for very good development of the project. If there is no devs interested in Nix dev env on macOS - why we keep checking it in every PR?

</details>

So, I propose:
  - [x] Introduce Hackage Cabal macOS test. And try to keep it mandatory.
  - [ ] Wait on the person that interested in supporting macOS actively. Over time, because of macOS specifics, projects would probably receive a "macOS operating mode", where projects would need to account that they are running on the macOS.  
  - [ ] Defer macOS Nix dev env from the patch merge check to `master` or once a week check. Hackage Cabal macOS test is considered enough, until there is a person interested in supporting Nix/pkgs on macOS.

2. We must decide what to do with Stackage maintenance

<details><summary>My previous yapping about it </summary>
<p>

>
> I do not have any exp with Stackage. I just guessed that here we had a management bottleneck here - there is from all evidence is also a bottleneck.
> 
> My own opinion (_do not hold anyone liable except me_):
> 
>   * That Stackage publish was a false-start. Projects are still under heavy development, Stackage is all about stability & reverse-compatibility, & our projects break that with every release. Our projects are still forming, the APIs are constantly changing. If we go Stackage - we would need to support backward compatibility or we would have Stackage freezes for stability and that causes additional problems & maintenance in the Nixpkgs.
> 
>   * I'd circumvented Stackage currently, as the projects are under development. We do not need to complicate things with it. Hackage releases then arrive directly to Nixpkgs.
> 
>   * And after all - what is the point of publishing 1 of 3 interlinked projects to the Stackage - to stuck divided in the middle between Hackage, Stackage & Nixpkgs?
> 
>   * Since 3 projects are interlinked and evolve together - they all at once should not be on Stackage - or all 3 go and support Stackage at once. And my own opinion - it still looks just a bit too early for that. HNix only now arrives at completing the core base of features, so to say - stability. & Especially before the HNix splitting into packages, Stackage and Nixpkgs dependency on it - it would complicate the splitting process, splitting in itself is already a big process that would pack a lot of much more important stuff, like whole API {reshuffle, clean-up, migration help}.
>     I'd voted to hold horses on Stackage. But also agree that trying to allow more people, throwing ... "resources" on it, sort to say, is also a universal way. But we do not have resources really.
>     Indeed, maybe if there were 6 people with access to the Stackage package - the situation probably'd resolved much faster.

</details>

So, I propose:
  - [x] kiss & yagni, Stackage currently seems not required & it is a platform for stabilized projects, publishing to it only complicates the maintenance process. We currently do not provide backward compatibility.
  - [ ] Go Stackage after the HNix API clean-up & split into multiple packages and we feel that the basic core situation stabilized and that we can claim&support Stackage stability and backward API compatibility.
    - [ ] If projects do Stackage - they all should go Stackage, because it makes sense only we all interlinked projects commit to it and published there.
      - [ ] Stackage complicates Nixpkgs maintenance. It is essentially an additional barrier in the downstream software flow into Nixpkgs. Nixpkgs imports projects directly from Hackage OR if they present on Stackage from Stackage, to not be caught in the middle - projects should be on Hackage or after that commit to moving all what is released to Stackage with hope that then Nixpkgs integration from it works.
  - [x] Team should limit what we support directly. I think after we coved the Haskell platforms + Nixpkgs - it would be enough, we need to draw a line there. We would need to live the distro package maintenance to downstreams - there are hundreds of repos, if someone wants to support some repo - it is "unofficial", because otherwise maintenance would be torn apart to what Debian/Ubuntu/Fedora/RHEL/Homebrew/Arch/Gentoo want.
  
---

*Formed from #104, #109, #111, https://github.com/commercialhaskell/stackage/issues/5766, https://github.com/NixOS/nixpkgs/pull/108010,  etc.*

**Comment by Anton-Latukha** [comment]
`layus` https://github.com/haskell-nix/hnix-store/issues/109#issuecomment-758507695
> ... I guess we can just disable that test on that platform until someone fixes it.

**Comment by Anton-Latukha** [comment]
I am satisfied with the current CI set.

I think the implicit changes are currently a good strut, project activity is low enough that we do not need to formalize the CI/support policies.
